### ISSUE TITLE
Pipes and Redirection

### ISSUE THREAD

**Issue by jorinvo** [trybash/bash-emulator#15]
We already have some chapters prepared for our [game](https://trybash.github.io/game/) to teach about pipes `|` and redirection`<` / `>`.

This can be implemented in a generic way in the `.run()` system method.

The command interface is already designed in a way to allow streaming of input and output.

- [x] `|`
- [ ] `<`
- [ ] `>`
- [ ] `>>`

**Comment by LordRatte** [comment]
Hi. Is there documentation on how to implement piping?

**Comment by jorinvo** [comment]
Hi @LordRatte! Thanks for your interest. Would you like to help adding pipes to bash-emulator?

To implement pipes, the [`run()`](https://github.com/trybash/bash-emulator/blob/gh-pages/src/index.js#L32) method needs to be modified to detect the `|` pipes in the input string and it would need to initialize multiple commands, chaining their output and input together.

Input is described [here](https://github.com/trybash/bash-emulator#writing-your-own-commands) but currently not implemented in any of the commands. 

The only existing command that can use input is `cat`. To add input support to `cat`, it would need to detect if there are no arguments passed and in that case return the `input` and `close` callbacks. 

To summarize, only `run` and `cat` need to be modified. After pipes are implemented you should be able to run a command like `cat myfile.txt | cat | cat`. 

Please let me know if you have any questions!

**Comment by LordRatte** [comment]
Hi, jorinvo. Sorry, I misunderstood. I thought you were saying that it was already available to be used. I would love to help out but I wouldn't even know how to start.

**Comment by jorinvo** [comment]
Good news! I added the pipes functionality. 
@LordRatte you can try it at https://trybash.github.io/bash-emulator/ 
with some command like `ls -l | cat -n`.

**Comment by LordRatte** [comment]
Hey, that's pretty cool @jorinvo . Thanks, man.
